COMETwebapp/Components/DefaultValues.razor

```diff
@@ -47,42 +47,211 @@ Copyright (c) 2022 RHEA System S.A.
+                            }
+                            Data.Add(new DataChart()
+                            {
+                                Value = nbDefaultValues,
+                                Argument = "Values as default",
+                                Domain = d.Name,
+                                Option = o.Name,
+                                State = s.Name,
+                                Parameters = parameters
+                            });
+                            Data.Add(new DataChart()
+                            {
+                                Value = parameterStates.Count - nbDefaultValues,
+                                Argument = "Other values",
+                                Domain = d.Name,
+                                Option = o.Name,
+                                State = s.Name,
+                                Parameters = parameterStates.FindAll(p => !parameters.Contains(p))
+                            });
+                            parameterOptions.RemoveAll(i => parameterStates.Contains(i));
+                        }));
+                        var nbDefaultValues = 0;
+                        var parameters = new List<ParameterValueSet>();
+                        if (ValuesType.Equals("Actual"))
+                        {
+                            parameters = parameterOptions.FindAll(i => i.ActualValue.First().Equals("-"));
+                            nbDefaultValues = parameters.Count;
+                        } else
+                        {
+                            parameters = parameterOptions.FindAll(i => i.Published.First().Equals("-"));
+                            nbDefaultValues = parameters.Count;
+                        }
+                        Data.Add(new DataChart()
+                            {
+                                Value = nbDefaultValues,
+                                Argument = "Values as default",
+                                Domain = d.Name,
+                                Option = o.Name,
+                                Parameters = parameters
+                            });
+                        Data.Add(new DataChart()
+                        {
+                            Value = parameterOptions.Count - nbDefaultValues,
+                            Argument = "Other values",
+                            Domain = d.Name,
+                            Option = o.Name,
+                            Parameters = parameterOptions.FindAll(p => !parameters.Contains(p))
+                        });
+                        parameterDomains.RemoveAll(i => parameterOptions.Contains(i));             
+                    });
+                    var nbDefaultValues = 0;
+                    var parameters = new List<ParameterValueSet>();
+                    if (ValuesType.Equals("Actual"))
+                    {
+                        parameters = parameterDomains.FindAll(i => i.ActualValue.First().Equals("-"));
```

Comment:
i doubt calling .First() is correct, it depends on whether this is a scalar or compound whether this is correct. In case the parameter references a CompoundParameterType some information will be missed